### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by hasnatali1947**
Features
- Role [ADMIN] can only delete comments.
- Role [ADMIN] can only delete posts.
- [ADMIN] can only see these buttons.

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    can :read, Comment, Public: true
+
+    return unless user.present?
+
+    can :read, :all
+
+    return unless user.role == 'admin'
+
+    can :manage, :all
+  end
```
Comment (abdulhamiid) [inline app/models/ability.rb]: According to the [project requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_add_authorization.md), A user can delete a `post` and a `comment` if it is theirs. please use CanCanCan for this authorization.  😊

For example:
```ruby
can :manage, Post, author: user`
```

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -26,6 +26,12 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+    @post.destroy
+    redirect_to root_path
+  end
+
```
Comment (abdulhamiid) [inline app/controllers/posts_controller.rb]: Please let's use **CanCanCan** for Post authorization in `Post Controller` 👍 

```ruby
  class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi Team 👋, 
Alt Text
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🌟

- ✔️ Limit access to web app resources based on authorization rules.
- ✔️ Use Git Flow
- ✔️ Meaningful commit messages.

## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -1,7 +1,9 @@
 class PostsController < ApplicationController
+  before_action :authenticate_user!, except: %i[index show]
+
```
Comment (rotimiazeez) [inline app/controllers/posts_controller.rb]: Please note that **CanCanCan** should be used for Post authorization in `Post Controller`, as pointed out by the previous CR. Kindly ensure that this is implemented 😉.
```
  class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
Hi @hasnatali1947 & @Roland-Ntwali  😉, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but some are still not addressed yet.

## Highlights
- Destroy actions are implemented in the PostsController and in the CommentsController. 👍🏽
- Descriptive PR ✌🏽
- Linter checks are passing 💪🏽

## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @hasnatali1947 , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] All linters passed
- [ ] cancancan ability used

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**